File: storeadmin/api/mutations/image/createProductImage.ts
import { imageFragment, productFragment } from '@/api/fragments'
import gql from 'graphql-tag'

export default gql`
  mutation createProductImage($uuid: ID!, $file: File!) {
    createProductImage(input: { uuid: $uuid, file: $file }) {
      product {
        ...productFragment
        images {
          ...imageFragment
        }
      }
      error
    }
  }
  ${imageFragment}
  ${productFragment}
`

File: server/app/graphql/mutations/image/create_product_image.rb
class Mutations::Image::CreateProductImage < Mutations::BaseMutation
  null false

  argument :uuid, ID, required: true
  argument :file, Types::FileType, required: false

  field :product, ObjectTypes::Product, null: true
  field :error, String, null: true

  def resolve(uuid:, file:)
    product = ::Product.find_by(uuid:)
    product.images.create!(image: file)

    { product: product.reload }
  rescue => e
    return GraphQL::ExecutionError.new(e.message)
  end
end
